### PR TITLE
feat: mode=silent

### LINKED ISSUE #26724: Support silent/full modes
### Describe the proposed change(s).

This requirement is primarily for use in the Mend app, but may also be useful to self-hosted scenarios too.

In the app we want to support three main concepts/modes:
- Silent: same as dryRun=lookup but also allows [checkedBranches](https://docs.renovatebot.com/self-hosted-configuration/#checkedbranches) to trigger the creation of branches/PRs. Any pre-existing branches should remain open, be updated, etc. Another way of thinking of this is like prCreation=approval but without the dependency dashboard issue. It could alternatively be named like "Scan only" or "On demand"
- Remediation: same as silent (allows checkedBranches) but defaults to prCreation=immediate for _remediation_ PRs, i.e. vulnerability fixes. This would be a first-class way to support the people who want "only security fixes" without needing to play with packageRules
- Full (default): same as today

The concept of dryRun would still exist and be compatible with and override the above. e.g. if -mode=silent and checkedBranches=renovate/foo and dryRun=lookup then the run wouldn't create any branch.

This mode should be user-configurable, e.g. if the user configures mode=remediation in renovate.json then it would override anything set by the app in global config.

### PR DESCRIPTION
**Pull request by rarkins**
## Changes

Adds a new config option "mode" with one new value: "silent".

## Context

This is built with the Mend Renovate App in mind. I also planned to add "remediation" as another supported mode, but will do that later.

Closes #26724

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/configuration-options.md [hunk 1]
```diff
@@ -2302,6 +2302,24 @@ This works because Renovate will add a "renovate/stability-days" pending status
 
 Add to this object if you wish to define rules that apply only to minor updates.
 
+## mode
```
Comment (rarkins) [inline docs/usage/configuration-options.md:2305]: Planned that the three modes will eventually be: full, silent, remediation

File: docs/usage/configuration-options.md [hunk 2]
```diff
@@ -2302,6 +2302,24 @@ This works because Renovate will add a "renovate/stability-days" pending status
...
+
+This configuration option was created primarily for use with Mend's hosted app, but can also be useful for some self-hosted use cases.
+
+It enables a new `silent` mode to allow repos to be scanned for updates _and_ for users to be able to request such updates be opened in PRs _on demand_ through the Mend UI, without needing the Dependency Dashboard issue in the repo.
```
Comment (rarkins) [inline docs/usage/configuration-options.md:2309]: I also thought of "on-demand" as an alternative name, but we use the term "silent" in our UI already so sticking to that.

**Review by secustor — COMMENTED** [review]
I don't like that we have multiple settings for such a modes. 

What prevents us to migrate dryRun to such a mode first? 

I would see the current dryRun implementation as `mode: dryRun` and then we can have a option like dryRunMode that would make it easier to understand IMO.

**Comment by rarkins** [comment]
> I don't like that we have multiple settings for such a modes.
> 
> What prevents us to migrate dryRun to such a mode first?

I thought about it, but dryRun is tangental to this. dryRun means "make no changes, just tell me what you'd do". mode=silent and future mode=remediation both can make changes, and they are also both compatible with dryRun (e.g. you could test how mode=silent behaves if you supply checkedBranches as a param)

> I would see the current dryRun implementation as `mode: dryRun` and then we can have a option like dryRunMode that would make it easier to understand IMO.

It's important to be able to combine the modes with dryRun, not make them exclusive like this. Plus you wouldn't save a config option either.

**Comment by secustor** [comment]
Hmm, than maybe should name it on-demand as you suggested in the issue. 🤔

As it is not really silent as it still uses checkedBranches.

**Comment by rarkins** [comment]
It's silent unless told otherwise. On-demand isn't perfect either because can be mixed up with the concept of Dependency Dashboard Approval.

**Review by viceice — DISMISSED** [review]

**Comment by rarkins** [comment]
Unfortunately this mode still clones for non-onboarded repos each time (even with repo cache enabled) but that's something we can improve later

File: lib/workers/repository/init/index.ts [hunk 3]
```diff
@@ -54,6 +54,11 @@ export async function initRepo(
   await initializeCaches(config as WorkerPlatformConfig);
   config = await getRepoConfig(config);
   setRepositoryLogLevelRemaps(config.logLevelRemap);
+  if (config.mode === 'silent') {
+    logger.info(
+      'Repository is running with mode=silent and will not make Issues or PRs by default',
+    );
```
Comment (rarkins) [inline lib/workers/repository/init/index.ts:60]: This INFO level messages when in silent mode should hopefully be more clear to users of the Mend app than currently where dryRun=lookup is debug and non-obvious

**Review by viceice — APPROVED** [review]

**Review by secustor — APPROVED** [review]

**Review by nabeelsaabna — APPROVED** [review]
LGTM
Note: any notifications we want to the user to know about will only appear in the UI (like deprecating encrypted token in the repository config)

**Pull request merged**